### PR TITLE
Genes and Case Ids selection

### PR DESCRIPTION
**Pull request by kalletlak**
Expose API's from iViz component to set/get genes and set case ids(patients and samples) from cbioportal.

Examples
iViz.session.manage.setGeneList(['TP53']);
iViz.session.manage.getGeneList();
iViz.session.manage.setSelectedPatients(['TCGA-BL-A0C8','TCGA-H4-A2HQ']);
iViz.session.manage.setSelectedSamples(['TCGA-DK-A3WX-01','TCGA-BL-A0C8-01']);

### CODE REVIEW

File: app/scripts/views/components/pieChart/pieChartTemplate.js [hunk 1]
```diff
@@ -74,14 +74,16 @@
...
-              this.chartInst.filter(temp);
+              this.chartInst.filter(newVal);
             }
           }else{
-            this.chartInst.filterAll();
+            if (newVal.length == 0) {
+              this.chartInst.filterAll();
+            }else{
+              var temp =newVal.length>1?[newVal]:newVal;
+              console.log(temp)
```
Comment (yichaoS) [inline app/scripts/views/components/pieChart/pieChartTemplate.js]: Is this for testing?

File: app/scripts/vueCore.js [hunk 2]
```diff
@@ -83,35 +83,93 @@
...
+                  var index = self_.selectedgenes.indexOf(gene);
+                  if(index === -1) {
+                    self_.selectedgenes.push(gene);
+                  }
+                  else{
+                    self_.selectedgenes.splice(index, 1);
+                  }
+                });
+              }
+              console.log(self_.selectedgenes);
```
Comment (yichaoS) [inline app/scripts/vueCore.js]: ?? testing line?

File: app/scripts/views/components/pieChart/pieChartTemplate.js (hunk 1 above)
Comment (kalletlak) [inline app/scripts/views/components/pieChart/pieChartTemplate.js]: Yeah!. Will cleanup that. Thanks for pointing

File: app/scripts/vueCore.js (hunk 2 above)
Comment (kalletlak) [inline app/scripts/vueCore.js]: This too...

File: app/scripts/views/components/pieChart/pieChartTemplate.js [hunk 3]
```diff
@@ -74,14 +74,16 @@
...
-              var exisitngFilters = $.extend(true, [],
-                this.chartInst.filters());
-              var temp = _.difference(exisitngFilters, newFilters);
-              this.chartInst.filter(temp);
+              this.chartInst.filter(newVal);
             }
           }else{
-            this.chartInst.filterAll();
+            if (newVal.length == 0) {
+              this.chartInst.filterAll();
```
Comment (yichaoS) [inline app/scripts/views/components/pieChart/pieChartTemplate.js]: ?? duplicate if newVal.length === 0??

**Pull request merged**